src/FullCalendar.tsx

```diff
@@ -1,32 +1,22 @@
 import * as React from 'react'
 import { Calendar, OptionsInput } from '@fullcalendar/core'
 
-export default class FullCalendar extends React.Component<OptionsInput, any> {
+const { forwardRef, useEffect, useImperativeHandle, useRef, useState } = React
 
-  private elRef: any = React.createRef()
-  private calendar: Calendar
+export default forwardRef((props: OptionsInput, componentRef) => {
+  const elRef = useRef()
+  const [calendar, setCalendar] = useState()
 
-  render() {
-    return (
-      <div ref={this.elRef}></div>
-    )
-  }
+  useEffect(() => {
+    const c = new Calendar(elRef.current, props)
+    setCalendar(c)
+    c.render()
+    return () => c.destroy()
+  }, [])
 
-  componentDidMount() {
-    this.calendar = new Calendar(this.elRef.current, this.props)
-    this.calendar.render()
-  }
+  useEffect(() => calendar && calendar.resetOptions(props), [calendar, props])
```

Comment:
can you avoid using these one-liner functions?

Reply:
Something like this?

```
  function init() {
    const c = new Calendar(elRef.current, props)
    setCalendar(c)
    c.render()
    // Cleanup callback automatically called before unmounting
    return () => c.destroy()
  }

  function updateOptions() {
    calendar && calendar.resetOptions(props);
  }

  function getApi(): Calendar {
    return calendar
  }

  // Calendar initialization
  useEffect(() => init(), [])

  // Calendar options update when props change
  useEffect(updateOptions, [calendar, props])

  // Allow functions to be exposed to parent component when using ref
  useImperativeHandle(componentRef, () => ({ getApi }), [calendar])
```

If you want going further avoiding one-liner functions:

```
  // Gather functions to be exposed with useImperativeHandle
  function createHandle() {
    return { getApi }
  }

  // Allow functions to be exposed to parent component when using ref
  useImperativeHandle(componentRef, createHandle, [calendar])
```

Reply:
I prefer something like this:

```js
// Calendar initialization
useEffect(() => {
  const c = new Calendar(elRef.current, props)
  setCalendar(c)
  c.render()
  // Cleanup callback automatically called before unmounting
  return () => c.destroy()
}, [])

// Calendar options update when props change
useEffect(() => {
  if (calendar) { // NOTE: i dont like using && operators as if statements
    calendar.destroy()
  }
}, [calendar, props])

// Allow functions to be exposed to parent component when using ref
useImperativeHandle(componentRef, () => {
  return {
    getApi() {
      return calendar
    }
  }
}, [calendar])
```